plugin/src/py/android_screenshot_tests/device_name_calculator.py

```diff
@@ -0,0 +1,70 @@
+from . import common
+import re
+
+
+class DeviceNameCalculator:
+    def __init__(self):
+        pass
+
+    def name(self):
+        api_version_text = self._api_version_text()
+        play_services_text = self._play_services_text()
+        screen_density_text = self._screen_density_text()
+        screen_size_text = self._screen_size_text()
+
```

Comment:
We should ensure that we have valid values here, and not `None`. If we receive `None` from any of the above functions then something is broken, so we should throw an exception here.